Sources/PackageLoading/ManifestLoader.swift

```diff
@@ -1137,15 +1142,28 @@ private final class SQLiteManifestCache: Closable {
     }
 
     func put(key: ManifestLoader.ManifestCacheKey, manifest: ManifestLoader.ManifestParseResult) throws {
-        let query = "INSERT OR IGNORE INTO MANIFEST_CACHE VALUES (?, ?);"
-        try self.executeStatement(query) { statement -> Void in
-            let data = try self.jsonEncoder.encode(manifest)
-            let bindings: [SQLite.SQLiteValue] = [
-                .string(key.sha256Checksum),
-                .blob(data),
-            ]
-            try statement.bind(bindings)
-            try statement.step()
+        do {
+            let query = "INSERT OR IGNORE INTO MANIFEST_CACHE VALUES (?, ?);"
+            try self.executeStatement(query) { statement -> Void in
+                let data = try self.jsonEncoder.encode(manifest)
+                let bindings: [SQLite.SQLiteValue] = [
+                    .string(key.sha256Checksum),
+                    .blob(data),
+                ]
+                try statement.bind(bindings)
+                try statement.step()
+            }
+        } catch (let error as StringError) where error.description == "database or disk is full" {
```

Comment:
SQLIte in TSC always throws StringErrors. We could change it to throw more structured errors to make this cleaner (move this logic there basically)

Reply:
Completely agree, but not needed as part of this PR, I'd think.

Reply:
https://github.com/apple/swift-tools-support-core/pull/179

Reply:
@abertelrud I think maybe we should do it in TSC first wdyt?

Reply:
I agree — the lower in the layers we can do this the better, and doing it in TSC now avoids the cleanup later.  Thanks for taking all this on!

Reply:
adjust to latest TSC